From 52248c856d1727f0e19fec757bf02508c03832da Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.dentz-von@nokia.com>
Date: Tue, 29 Jun 2010 15:26:22 +0300
Subject: [PATCH 18/37] Fix use of errno after calling some function that could overwrite it

---
 src/device.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/device.c b/src/device.c
index ea0640e..6af76d1 100644
--- a/src/device.c
+++ b/src/device.c
@@ -1891,8 +1891,9 @@ static int device_authentication_requested(struct btd_device *device,
 
 	dd = hci_open_dev(adapter_get_dev_id(device->adapter));
 	if (dd < 0) {
-		error("Unable to open adapter: %s(%d)", strerror(errno), errno);
-		return -errno;
+		int err = -errno;
+		error("Unable to open adapter: %s(%d)", strerror(-err), -err);
+		return err;
 	}
 
 	memset(&rp, 0, sizeof(rp));
@@ -1910,10 +1911,11 @@ static int device_authentication_requested(struct btd_device *device,
 	rq.event  = EVT_CMD_STATUS;
 
 	if (hci_send_req(dd, &rq, HCI_REQ_TIMEOUT) < 0) {
+		int err = -errno;
 		error("Unable to send HCI request: %s (%d)",
-					strerror(errno), errno);
+					strerror(-err), -err);
 		hci_close_dev(dd);
-		return -errno;
+		return err;
 	}
 
 	if (rp.status) {
-- 
1.7.2.3

